e2e_tests/tests/command/test_tensorboard.py

```diff
@@ -32,14 +32,23 @@ def shared_fs_config(num_trials: int) -> str:
 """
 
 
-def s3_config(num_trials: int, secrets: Dict[str, str]) -> str:
-    return f"""
+def s3_config(num_trials: int, secrets: Dict[str, str], prefix: Union[str, None] = None) -> str:
```

Comment:
Please use `Optional[str]` instead of `Union[str, None]`.  They're equivalent in meaning but `Optional[str]` is the canonincal option.